AsyncDisplayKit/Details/ASTraitCollection.h

```diff
@@ -0,0 +1,29 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <UIKit/UIKit.h>
+#import <AsyncDisplayKit/ASEnvironment.h>
+
+@interface ASTraitCollection : NSObject
+
+@property (nonatomic, assign) BOOL isMutable;
```

Comment:
I'd recommend dropping this in favor of making the ASTraitCollection immutable the moment it's initialized. Will require a larger initializer, but will reduce potential mistakes.